[28.x] Further backports and rc2

achow101 opened the pull request:
* #30834
* #30807
* #30880
* https://github.com/bitcoin-core/gui/pull/835
* #30899
* #30884

furszy committed:
test: add coverage for assumeUTXO honest peers disconnection

Exercising and verifying the following points:

1. An IBD node can sync headers from an AssumeUTXO node at
   any time.

2. IBD nodes do not request historical blocks from AssumeUTXO
   nodes while they are syncing the background-chain.

3. The assumeUTXO node dynamically adjusts the network services
   it offers according to its state.

4. IBD nodes can fully sync from AssumeUTXO nodes after they
   finish the background-chain sync.

Github-Pull: bitcoin/bitcoin#30807
Rebased-From: 992f83bb6f4b29b44f4eaace1d1a2c0001d43cac

furszy committed:
assumeUTXO: fix peers disconnection during sync

Because AssumeUTXO nodes prioritize tip synchronization, they relay their local
address through the network before completing the background chain sync.
This, combined with the advertising of full-node service (NODE_NETWORK), can
result in an honest peer in IBD connecting to the AssumeUTXO node (while syncing)
and requesting an historical block the node does not have. This behavior leads to
an abrupt disconnection due to perceived unresponsiveness (lack of response)
from the AssumeUTXO node.

This lack of response occurs because nodes ignore getdata requests when they do
not have the block data available (further discussion can be found in PR 30385).

Fix this by refraining from signaling full-node service support while the
background chain is being synced. During this period, the node will only
signal 'NODE_NETWORK_LIMITED' support. Then, full-node ('NODE_NETWORK')
support will be re-enabled once the background chain sync is completed.

Github-Pull: bitcoin/bitcoin#30807
Rebased-From: 6d5812e5c852c233bd7ead2ceef051f8567619ed

achow101 committed:
build: Bump to 28.0rc2

achow101 committed:
doc: Generate manpages

Commit:
test: Work around boost compilation error

Github-Pull: bitcoin/bitcoin#30834
Rebased-From: fa9d7d5d205ada8915cbbc29599ab8e7bf1fffe0

fanquake commented on src/txmempool.h:
```diff
@@ -23,7 +23,17 @@
 #include <util/result.h>
 #include <util/feefrac.h>
 
+// This works around a bug in Boost <= 1.80.0 when using Clang >=18.
+// See https://github.com/bitcoin/bitcoin/issues/30751.
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wenum-constexpr-conversion"
+#endif
```
This can now be swapped out for #30834.

achow101 replied:
Backported fa9d7d5d205ada8915cbbc29599ab8e7bf1fffe0

hebasto reviewed (approved):
ACK 37a6739203ea3da394acef38b18a794e67543a90.

furszy reviewed (commented):
utACK b97323c7aaa06bda8c76dfc754c90f9de44f4898

fjahr committed:
test: Wait for local services to update in feature_assumeutxo

Github-Pull: bitcoin/bitcoin#30880
Rebased-From: 19f4a7c95a99162122068d4badffeea240967a65

sipa committed:
streams: cache file position within AutoFile

Github-Pull: bitcoin/bitcoin#30884
Rebased-From: e624a9bef16b6335fd119c10698352b59bf2930a

furszy committed:
gui: fix crash when closing wallet

The crash occurs because 'WalletController::removeAndDeleteWallet' is called
twice for the same wallet model: first in the GUI's button connected function
'WalletController::closeWallet', and then again when the backend emits the
'WalletModel::unload' signal.

This causes the issue because 'removeAndDeleteWallet' inlines an
erase(std::remove()). So, if 'std::remove' returns an iterator to the end
(indicating the element wasn't found because it was already erased), the
subsequent call to 'erase' leads to an undefined behavior.

Github-Pull: bitcoin-core/gui#835
Rebased-From: a965f2bc07a3588f8c2b8d6a542961562e3f5d0e

hebasto commented:
@achow101 

Could you please add https://github.com/bitcoin-core/gui/pull/835 as well?

achow101 commented:
> @achow101
> 
> Could you please add [bitcoin-core/gui#835](https://github.com/bitcoin-core/gui/pull/835) as well?

Done

hebasto committed:
qt: Translations update

The recent translations from Transifex.com 28.x fetched with the
bitcoin-maintainer-tools/update-translations.py tool.

Github-Pull: bitcoin/bitcoin#30899
Rebased-From: ae0529576147a1a5bee992574e2cefc8a1fa37d0

hebasto commented:
@achow101 

Feel free to pick [this](https://github.com/hebasto/bitcoin/commit/be6354db410d162f82b5e3c76d964c0365e51d96) translation update, which resolves https://github.com/bitcoin/bitcoin/issues/30897 and includes new translations made over the nearly 3 weeks since https://github.com/bitcoin/bitcoin/pull/30715.

fanquake commented:
> Feel free to pick [this](https://github.com/hebasto/bitcoin/commit/be6354db410d162f82b5e3c76d964c0365e51d96) translation update

There's not so much of a rush here that this should bypass the normal PR & review process. This should first be PR'd and merged into master and then backported. Otherwise any broken string are still in master. There is at least one other change waiting for 28.x, so rc2 also needs to either wait for that, or we'll be having an rc3 (which can include the ts update) in any case.

hebasto commented:
> There's not so much of a rush here that this should bypass the normal PR & review process.

Sure.

> This should first be PR'd and merged into master and then backported.

After branching off, the Transifex.com has translation for the version branches only. Fetching 28x translations into the master branch does not look reasonable.

> Otherwise any broken string are still in master. 

Maybe this case is a special one.

hebasto commented:
> > There's not so much of a rush here that this should bypass the normal PR & review process.
> 
> Sure.

Please see https://github.com/bitcoin/bitcoin/pull/30899.

sipa committed:
streams: remove AutoFile::Get() entirely

Co-Authored-By: David Gumberg <davidzgumberg@gmail.com>

Github-Pull: bitcoin/bitcoin#30884
Rebased-From: a240e150e837b5a95ed19765a2e8b7c5b6013f35

stickies-v reviewed (approved):
ACK 06a7df70df30879e0b691d1a252636f703b8cdfb

I verified that all backports are clean, except:
- b329ed739b7311b3b47cae1ef8d576a90e0a36a1 backported from 992f83bb6f4b29b44f4eaace1d1a2c0001d43cac: merge conflict due to a difference in `sha256sum_file` import, trivially resolved

I didn't review each commit in-depth since they're clean, but the changes backported seem sensible.

hebasto reviewed (approved):
ACK 06a7df70df30879e0b691d1a252636f703b8cdfb, I've backported the listed PRs locally. The only merge conflict I faced was in https://github.com/bitcoin/bitcoin/pull/30807. It was trivial to resolve.
